src/Oliver.Common/Infrastructure/FileLoggerProvider.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Oliver.Common.Infrastructure
+{
+    public class FileLoggerProvider : ILoggerProvider
+    {
+        private readonly string fileName;
+        private readonly int maxLogSizeMb;
+        private readonly int maxFilesCount;
+        private readonly string archiveLogFileFormat;
+        private readonly object locker = new object();
```

Comment:
The latest fix hasn't resolved the issue cause we still can create 2 or more loggers over the same file with different locks.

Reply:
done